### PR TITLE
issue-844

### PR DESCRIPTION
**Pull request by kngwyu**
For #844 
Now what I did is only skip glob after we construct a path from glob and 2 tests fail.
I'll fix tests by this weekend.

### CODE REVIEW

**Comment by kngwyu** [comment]
I decided to limit the recursion depth via glob to 4.
Before:
```
test glob_imports5 ... bench:  53,956,377 ns/iter (+/- 2,045,050)
test glob_imports6 ... bench: 307,321,133 ns/iter (+/- 34,592,106)
test glob_imports7 ... bench: 2,046,519,408 ns/iter (+/- 161,600,731)
```
After
```
test glob_imports5 ... bench:  44,205,820 ns/iter (+/- 2,150,180)
test glob_imports6 ... bench: 107,402,297 ns/iter (+/- 3,073,601)
test glob_imports7 ... bench: 217,409,533 ns/iter (+/- 5,366,173)
```

It's still slow(maybe we should cache the result of `resolve_path`), but much better.
However, this PR means we give up trying to complete the case where we have to go through the blob imports more than 5 times, e.g.
```
    use mod1::*; use mod2::*; use mod3::*; use mod4::*;
    use Bar::*;
    mod mod1 { pub mod mod2 { pub mod mod3 { pub mod4 {
        pub enum Bar { MyVariant, MyVariant2 }
    }}}}
    MyVa~riant
```

File: src/racer/matchers.rs [hunk 1]
```diff
@@ -12,13 +12,28 @@ use std::{str, vec};
...
 }
 
 /// A stack of imports (`use` items) currently being resolved.
-pub type PendingImports<'stack, 'fp> = StackLinkedListNode<'stack, PendingImport<'fp>>;
+type PendingImports<'stack, 'fp> = StackLinkedListNode<'stack, PendingImport<'fp>>;
+
+/// Import information(pending imports, glob, and etc.)
+pub struct ImportInfo<'stack, 'fp: 'stack> {
+    imports: PendingImports<'stack, 'fp>,
+    glob_depth: Option<usize>,
```
Comment (TedDriggs) [inline src/racer/matchers.rs]: Can you add a doc comment to this please?

**Comment by kngwyu** [comment]
Thanks for review, done.

**Comment by kngwyu** [comment]
Changed max depth to 3 from 4 in https://github.com/racer-rust/racer/commit/db927a7e946554ea3c57f02f8d28d453f465c054

**Pull request merged**